modules/core/src/va_intel.cpp

```diff
@@ -608,8 +608,32 @@ void convertToVASurface(VADisplay display, InputArray src, VASurfaceID surface,
 
         VAImage image;
         status = vaDeriveImage(display, surface, &image);
-        if (status != VA_STATUS_SUCCESS)
-            CV_Error(cv::Error::StsError, "VA-API: vaDeriveImage failed");
+        if (status != VA_STATUS_SUCCESS){
+            //try vaCreateImage + vaPutImage
+            //pick a format
+            int num_formats = vaMaxNumImageFormats(display);
+            VAImageFormat fmt_list[num_formats];
+            status = vaQueryImageFormats(display, fmt_list, &num_formats);
+            if (status != VA_STATUS_SUCCESS)
+                CV_Error(cv::Error::StsError, "VA-API: vaQueryImageFormats failed");
+            VAImageFormat *selected_format = &fmt_list[0];
+            for (int i = 0; i < num_formats; i++){
+                if (fmt_list[i].fourcc == VA_FOURCC_NV12 || fmt_list[i].fourcc == VA_FOURCC_YV12){
+                    selected_format = &fmt_list[i];
+                    break;
+                }
+            }
+
+            status = vaCreateImage(display, selected_format, size.width, size.height, &image);
```

Comment:
What if `selected_format` is not "found"? Why do we fallback to the first one?

I believe we should emit error message as fast as possible.